diffrax/integrate.py

```diff
@@ -142,34 +218,107 @@ def body_fun(state, inplace):
                 state.made_jump,
             )
         else:
-            # TODO: double-check that these are the correct `ts_size` and
-            # `direction`.
-            history = DenseInterpolation(
+            min_delay = []
+            flat_delays = jtu.tree_leaves(delays.delays)
+            for delay in flat_delays:
+                min_delay.append(delay(state.tprev, state.y, args))
+            min_delay = jnp.stack(min_delay).min()
+            implicit_step = min_delay < state.tnext - state.tprev
```

Comment:
Let's have some brackets here: `min_delay < (state.tnext - state.tprev)`. I honestly don't know the precedence rules here off the top of my head... so for readability it's better to be clear that this is computing the right thing.

Reply:
done